### PR TITLE
feat(skymp5-server): prevent players from using attribute regeneration potions very often 

### LINKED ISSUE #1278: Prevent players from using attribute regeneration potions very often
Currently, a winner of a battle between players usually depends on a person who has more regenerative potions, which is not really good. A player can spam the such potion and, thus, fights turns to a boring activity. We do not want that. So, a solution was to forbid players abuse the present battle mechanics.

### PR DESCRIPTION
**Pull request by che6a**
Closes #1278

### CODE REVIEW

**Review by Pospelove — CHANGES_REQUESTED** [review]
Should be enabled only for servers with sweetpie.esp in mods list

plus we usually do not use `using namespace std::chrono_literals;`

**Comment by che6a** [comment]
> Should be enabled only for servers with sweetpie.esp in mods list
> 
> plus we usually do not use `using namespace std::chrono_literals;`

Yeah, I thought about these. I'll fix it later today

**Comment by nic11** [comment]
> plus we usually do not use `using namespace std::chrono_literals;`

Why not?

**Comment by Pospelove** [comment]
> Why not?
well usually we do not, we might want to change that but in the current codebase we do not use this feature

**Review by nic11 — APPROVED** [review]

**Comment by Pospelove** [comment]
Are you sure there is SWEETPIE switch??

File: skymp5-server/cpp/server_guest_lib/MpActor.cpp [hunk 1]
```diff
@@ -398,17 +409,51 @@ void MpActor::EatItem(uint32_t baseId, espm::Type t)
   } else {
     return;
   }
-
+  std::unordered_set<std::string> modFiles = { GetParent()->espmFiles.begin(),
+                                               GetParent()->espmFiles.end() };
+  bool hasSweetpie = modFiles.count("SweetPie.esp");
```
Comment (Wedmer) [inline skymp5-server/cpp/server_guest_lib/MpActor.cpp:414]: I think this is useful not only for sweetpie. Maybe redo this as optional configuration parameters?

**Review by Wedmer — CHANGES_REQUESTED** [review]

File: skymp5-server/cpp/server_guest_lib/MpActor.cpp (hunk 1 above)
Comment (Pospelove) [reply skymp5-server/cpp/server_guest_lib/MpActor.cpp:414]: using such check is actually acceptable. there is already such check in codebase so it's pretty consistent.

the ideal solution would be to implement this on gamemode side

**Review by Pospelove — APPROVED** [review]

**Comment by Pospelove** [comment]
todo: merge after ci is green

**Pull request merged**